pkg/node_reputation/node_reputation_handler.go

```diff
@@ -0,0 +1,280 @@
+// Copyright (C) 2018 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package nodereputation
+
+import (
+	"database/sql"
+	"strconv"
+
+	"golang.org/x/net/context"
+	//"storj.io/storj/protos/node_reputation"
+	proto "storj.io/storj/protos/node_reputation"
+)
+
+// Server is a struct
+type Server struct{}
+
+// UpdateReputation in handler
+func (s *Server) UpdateReputation(ctx context.Context, in *proto.NodeUpdate) (*proto.UpdateReply, error) {
+	db, err := SetServerDB("./Server.db")
+	if err != nil {
+		return nil, err
+	}
+
+	status := insertNodeUpdate(db, in)
+
+	return &proto.UpdateReply{
+		BridgeName: "Storj",
+		NodeName:   in.NodeName,
+		Status:     status,
+	}, nil
+}
+
+// NodeReputation in handler
+func (s *Server) NodeReputation(ctx context.Context, in *proto.NodeQuery) (*proto.NodeReputationRecord, error) {
+	db, err := SetServerDB("./Server.db")
+	if err != nil {
+		return nil, err
+	}
+	node, err := byNodeName(db, in.NodeName)
+	if err != nil {
+		return nil, err
+	}
+
+	return &node, nil
+}
+
+// FilterNodeReputation in handler
+func (s *Server) FilterNodeReputation(ctx context.Context, in *proto.NodeFilter) (*proto.NodeReputationRecords, error) {
+	db, err := SetServerDB("./Server.db")
+	if err != nil {
+		return nil, err
+	}
+	nodes, err := selectNodeWhere(db, in.ColumnName, in.Operand, in.ColumnValue)
+	if err != nil {
+		return nil, err
+	}
+
+	return &nodes, nil
+}
+
+// PruneNodeReputation compresses a node's reputation
+func (s *Server) PruneNodeReputation(ctx context.Context, in *proto.NodeQuery) (*proto.UpdateReply, error) {
+	db, err := SetServerDB("./Server.db")
+	if err != nil {
+		return nil, err
+	}
+
+	status := deleteByNodeName(db, in.NodeName)
+
+	return &proto.UpdateReply{
+		BridgeName: "Storj",
+		NodeName:   in.NodeName,
+		Status:     status,
+	}, nil
+
+}
+
+// byNodeName function used in handler by update reputation
+func byNodeName(db *sql.DB, nodeName string) (proto.NodeReputationRecord, error) {
+	var recordForError proto.NodeReputationRecord
+	selectNodeStmt := genWhereStatement(selectAllStmt, nodeNameColumn, equal, nodeName)
+	row, err := endianReputation(db, selectNodeStmt)
+	if err != nil {
+		return recordForError, err
+	}
+
+	return row.serde(row.naiveScore()), nil
+}
+
+// deleteByNodeName compresses a node's reputation
+func deleteByNodeName(db *sql.DB, nodeName string) proto.UpdateReply_ReplyType {
```

Comment:
hmm maybe i misunderstood this earlier. are there multiple rows per node id? if so, how come?